Orange/widgets/utils/colorpalette.py

```diff
@@ -597,66 +597,52 @@ def __getitem__(self, val):
-            self.set_number_of_colors(max(rgb_colors.keys()))
-        else:
-            self.rgb_colors_dict = None
-            self.set_number_of_colors(number_of_colors)
+            number_of_colors = max(rgb_colors.keys())
+        self.set_number_of_colors(number_of_colors)
 
-    def set_number_of_colors(self, number_of_colors=None):
+    def set_number_of_colors(self, number_of_colors=0):
         """Change the palette if there are palettes for different number of
-           colors. Else, just copy rgbColors to numpy array"""
+           colors. Else, just copy colors as numpy array"""
         self.number_of_colors = number_of_colors
-        if self.rgb_colors_dict is not None:
+        if isinstance(self.rgb_colors, dict):
             number_of_colors = max(3, number_of_colors)
-            if number_of_colors not in self.rgb_colors_dict:
-                keys = [n for n in self.rgb_colors_dict
-                        if n >= number_of_colors]
-                if keys:
-                    number_of_colors = min(keys)
-                else:
+            if number_of_colors not in self.rgb_colors:
+                try:
+                    number_of_colors = min([n for n in self.rgb_colors
+                                            if n >= number_of_colors])
+                except ValueError:
                     raise ValueError("Not enough colors")
-
-            self.rgb_colors = \
-                self.rgb_colors_dict[number_of_colors]
-
-        elif number_of_colors is None or \
-                number_of_colors < len(self.rgb_colors):
-            self.rgb_colors = self.default_colors
+            rgb_colors = self.rgb_colors[number_of_colors]
+        elif number_of_colors <= len(self.rgb_colors):
+            rgb_colors = self.rgb_colors
         else:
-            self.rgb_colors = []
+            rgb_colors = []
             for i in range(self.number_of_colors):
                 col = QColor()
                 col.setHsv(360 / number_of_colors * i, 255, 255)
-                self.rgb_colors.append(col.getRgb()[:3])
-        self.rgb_array = np.array(self.rgb_colors)
-        self.rgba_array = np.hstack([
-            self.rgb_array, np.full((len(self.rgb_colors), 1), 255)])
+                rgb_colors.append(col.getRgb()[:3])
+        self.rgb_array = np.vstack((rgb_colors, [NAN_GREY]))
 
     def __getitem__(self, index):
-        return QColor(*self.getRGB(index))
-
-    # noinspection PyPep8Naming
-    def getRGB(self, index):
-        if isinstance(index, np.ndarray):
-            return self.rgb_array[index.astype(int)]
-        else:
-            return self.rgb_colors[int(index)]
-
-    # noinspection PyPep8Naming
-    def getRGBA(self, index):
-        if isinstance(index, np.ndarray):
-            return self.rgba_array[index.astype(int)]
+        if isinstance(index, Number):
+            return QColor(*self.getRGB(index))
+        return [QColor(*c) for c in self.getRGB(index)]
```

Comment:
The old version always returned a single QColor for a given index.
When extending functionality I find it better to leave the previous way the default one.

In this case that means check for the new special case first
`if isinstance(index, Iterable): return [...]`
and in all other cases retain the previous functionality.

In your case when index is not a Number and not iterable it crashes in a new way...